### PR TITLE
feat(k8s): Add Deployment Kind support for Blue/Green deployments

### PR DESCRIPTION
**Pull request by ciurescuraul**
Update Clouddriver to support Deployment kind for Red/Black (deprecated) deployments, which will become Blue/Green.

### CODE REVIEW

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]
LGTM, but let's add unit tests (if possible) and, more important, one or more integration tests. 

see `clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/DeployManifestIT.java`

File: clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/DeployManifestIT.java
Comment (ovidiupopa07) [inline clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/DeployManifestIT.java:1360]: let's also add a test for the new `blue-green` traffic management strategy

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]

**Comment by ciurescuraul** [comment]
1. Deploy (Manifest)
image

2. Disable (Manifest)
image

3. Pods after running the pipeline
image

4. LoadBalancer redirects traffic to newly created pods
image

5. LoadBalancer service description
image

File: clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/DeployManifestIT.java
Comment (ciurescuraul) [reply clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/DeployManifestIT.java]: addressed

**Review by ovidiupopa07 — APPROVED** [review]
LGTM if the tests are passing

File: clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/KubernetesDeploymentHandler.java
Comment (dbyron-sf) [inline clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/KubernetesDeploymentHandler.java:89]: This change breaks a bunch of integration tests (which unfortunately weren't required to pass before merging this PR), but also is a pretty massive behavior change.  Before people who deploy a deployment named `foo`, would get `foo`.  Now they get `foo-vNNN`.  It also makes https://spinnaker.io/docs/reference/providers/kubernetes-v2/#workloads incorrect for Deployments.  I'm not sure how critical this change is to this PR....

Comment (dbyron-sf) [reply clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/KubernetesDeploymentHandler.java:89]: PR to revert this one: https://github.com/spinnaker/clouddriver/pull/5843

**Pull request merged**
